dds/DCPS/XTypes/DynamicData.h

```diff
@@ -304,6 +309,10 @@ class OpenDDS_Dcps_Export DynamicData {
   ACE_CDR::ULong item_count_;
 };
 
+bool OpenDDS_Dcps_Export print_dynamic_data(
+  OpenDDS::XTypes::DynamicData dd, OpenDDS::XTypes::DynamicType_rch dt,
+  OpenDDS::DCPS::String& type_string, OpenDDS::DCPS::String indent);
+
```

Comment:
If each DynamicData knows its type, is the second parameter needed?
Does DynamicData need to be passed by `const &`?
Should this be part of the library or part of the application?

Reply:
1. No it is not, I will remove it.
2. No, that would cause errors when calling the other DynamicData functions like `get_*_value`.
3. I put it in the library so I could have both the app and the test both use the same code.

Reply:
2. then pass by reference - we don't want a copy